projects/Douglas Mentoring/day1.py

```diff
@@ -0,0 +1,92 @@
+import getpass
+import netmiko
+# json was here bc as I'm using textfsm to pull data from cisco devices
+# it comes in json i needed to show the output before deciding how to tread the date
+import json
+import csv
+
+# Where i'm pulling the devices list / Soon I'll make this automated which will pull direct from NARF
+file = open(r'/home/segattod/segattod-workspace/src/Segattod-package/myproject/Light_level_checks/list.txt',
+            'r')
+file = file.read().splitlines()
+
+username = input('User Name:')
+password = getpass.getpass()
+
+# Variable for "append" the data later
+formatted_data = []
+
+# for now just to name the file / Soon to add for user to add input :)
+whid = 'YEG1'
+
+
+# Function to sorte the data
+def light_level_check(connection):
+    # Command to checks light level per interface
+    transceiver_output = connection.send_command('show interface transceiver', read_timeout=90,
+                                                 use_textfsm='~/ntc-templates/ntc_templates/templates/cisco_ios_show_interface_transceiver.textfsm')
```

Comment:
Looking at the code for `send_command`, `use_textfsm`is a boolean so it expects True or False.  I suspect you meant to use `textfsm_template` there?

You didn't get any errors here because Python is a "Truthy" language.  When it expects a boolean and it gets pretty much anything, it assumes that is True.  It's smart enough to know that an empty string is False, and the integer 0 is false, but the string "banana" or any number other than 0 is treated as a True.

Reply:
not sure if i get your question right, but zthe output from this command will be formatted as the textfsm template, which should look just like json. 

here is the output how it looks like (ps: i added just 2 lines/interfaces output but it could have many)

```
output = [
{
    "iface": "Te1/0/39",
    "temperature": "36.1",
    "voltage": "3.29",
    "tx_pwr": "41.8",
    "rx_pwr": "-2.5"
  },
  {
    "iface": "Te1/0/40",
    "temperature": "35.1",
    "voltage": "3.30",
    "tx_pwr": "36.5",
    "rx_pwr": "-2.1"
  }
]
```

Reply:
https://ktbyers.github.io/netmiko/docs/netmiko/base_connection.html#netmiko.base_connection.BaseConnection.send_command
![image](https://github.com/ferruzzi/scrapheap/assets/1920178/9205fcfa-dd2b-4d75-848b-d643a7514907)

In their documentation, they say that `use_textfsm` is a boolean (True or False) and it defaults to False.  So your string `'~/ntc-templates/ntc_templates/templates/cisco_ios_show_interface_transceiver.textfsm'` that you have there is only being read as "True".  I think what you meant was:

```suggestion
    transceiver_output = connection.send_command(
            'show interface transceiver', 
            read_timeout=90,
            use_textfsm=True,
            textfsm_template= '~/ntc-templates/ntc_templates/templates/cisco_ios_show_interface_transceiver.textfsm'
       )
```